addons/product_extended/product_extended.py

```diff
@@ -85,31 +63,26 @@ def _calc_price(self, cr, uid, bom, test = False, real_time_accounting=False, co
                 cycle = wline.cycle_nbr
                 hour = (wc.time_start + wc.time_stop + cycle * wc.time_cycle) *  (wc.time_efficiency or 1.0)
                 price += wc.costs_cycle * cycle + wc.costs_hour * hour
-                price = self.pool.get('product.uom')._compute_price(cr,uid,bom.product_uom.id, price, bom.product_id.uom_id.id)
+                price = ProductUom._compute_price(bom.product_uom.id, price, bom.product_id.uom_id.id)
         
         #Convert on product UoM quantities
         if price > 0:
-            price = uom_obj._compute_price(cr, uid, bom.product_uom.id, price / bom.product_qty, bom.product_id.uom_id.id)
+            price = ProductUom._compute_price(bom.product_uom.id, price / bom.product_qty, bom.product_id.uom_id.id)
 
-        product = tmpl_obj.browse(cr, uid, bom.product_tmpl_id.id, context=context)
+        product = self.browse(bom.product_tmpl_id.id)
         if not test:
             if (product.valuation != "real_time" or not real_time_accounting):
-                tmpl_obj.write(cr, uid, [product.id], {'standard_price' : price}, context=context)
+                product.write({'standard_price' : price})
             else:
                 #Call wizard function here
-                wizard_obj = self.pool.get("stock.change.standard.price")
-                ctx = context.copy()
-                ctx.update({'active_id': product.id, 'active_model': 'product.template'})
-                wiz_id = wizard_obj.create(cr, uid, {'new_price': price}, context=ctx)
-                wizard_obj.change_price(cr, uid, [wiz_id], context=ctx)
+                wiz_id = self.env["stock.change.standard.price"].create({'new_price': price})
+                wiz_id.change_price()
         return price
 
 
-class product_bom(osv.osv):
+class ProductBom(models.Model):
     _inherit = 'mrp.bom'
             
-    _columns = {
-        'standard_price': fields.related('product_tmpl_id','standard_price',type="float",relation="product.product",string="Standard Price",store=False)
-    }
+    standard_price = fields.Float(related='product_tmpl_id.standard_price', relation="product.product", string="Standard Price", store=False)
```

Comment:
IIRC `store=False` is the default for computed and related fields.

Also the `relation` parameter makes no sense, remove it.